presto-local-file/src/main/java/com/facebook/presto/localfile/LocalFileRecordCursor.java

```diff
@@ -0,0 +1,323 @@
+    }
+
+    @Override
+    public void close()
+    {
+    }
+
+    private static class FilesReader
+    {
+        private final Iterator<File> files;
+        private final Optional<Domain> domain;
+        private final OptionalInt timestampOrdinalPosition;
+
+        private BufferedReader reader;
+
+        public FilesReader(OptionalInt timestampOrdinalPosition, Iterator<File> files, TupleDomain<LocalFileColumnHandle> predicate)
+                throws IOException
+        {
+            requireNonNull(files, "files is null");
+            checkArgument(files.hasNext(), "files is empty");
+
+            this.files = files;
+            createNextReader();
+
+            requireNonNull(predicate, "predicate is null");
+
+            Optional<Map<LocalFileColumnHandle, Domain>> domains = predicate.getDomains();
+            Domain domain = null;
+            if (domains.isPresent() && timestampOrdinalPosition.isPresent()) {
+                Map<LocalFileColumnHandle, Domain> domainMap = domains.get();
+                Set<Domain> timestampDomain = domainMap.entrySet().stream()
+                        .filter(entry -> entry.getKey().getOrdinalPosition() == timestampOrdinalPosition.getAsInt())
+                        .map(Map.Entry::getValue)
+                        .collect(toSet());
+
+                if (!timestampDomain.isEmpty()) {
+                    domain = Iterables.getOnlyElement(timestampDomain);
+                }
+            }
+            this.domain = Optional.ofNullable(domain);
+            this.timestampOrdinalPosition = timestampOrdinalPosition;
+        }
+
+        private void createNextReader()
+                throws IOException
+        {
+            File file = files.next();
+            FileInputStream fileInputStream = new FileInputStream(file);
+            InputStream in = (file.getName().endsWith(".gz")) ? new GZIPInputStream(fileInputStream) : fileInputStream;
+            this.reader = new BufferedReader(new InputStreamReader(in));
+        }
+
+        public String readLine()
+                throws IOException
+        {
+            String line = null;
+            boolean newReader = false;
+            while (line == null) {
+                line = reader.readLine();
+                if (line != null) {
+                    if (!newReader || meetsPredicate(line)) {
+                        return line;
+                    }
+                }
+                if (!files.hasNext()) {
+                    return null;
+                }
+                createNextReader();
+                newReader = true;
+            }
+            return line;
+        }
+
+        private boolean meetsPredicate(String line)
+        {
+            if (!timestampOrdinalPosition.isPresent() || !domain.isPresent()) {
+                return true;
+            }
+
+            List<String> fields = LINE_SPLITTER.splitToList(line);
```

Comment:
since we need to split the line when there is a filter on timestamp ordinal, maybe change the reader to return a List<String> so we don't have to split twice